File: beecrowd/1804.cpp
#include<bits/stdc++.h>
using namespace std;

#define MAX 123456
typedef long long ll;


int n, bit[MAX], nlis[MAX];

int setbit(int pos, int diff){
    while (pos <= n){
        bit[pos] += diff;
        pos += pos & -pos;
    }
}

int getbit(int pos){
    int ans = 0;
    while (pos){
        ans += bit[pos];
        pos -= pos & -pos;
    }
    return ans;
}

int main(void){
    int i, x, c, d;
    ll ans = 0;
    scanf("%d", &n);
    memset(bit, 0, sizeof(bit));
    for (i = 1; i <= n; i++){
        scanf("%d", &x);
        nlis[i] = x;
        setbit(i, x);
    }
    while (scanf("%c %d", &d, &c) != EOF){
        if (d == 'a'){
            setbit(c, -nlis[c]);
        }
        if (d == '?'){
            int ans = getbit(c-1);
            printf("%d", ans);
        }
    }
}

File: beecrowd/1804.py
nums = int(input())
bit = [0] * (nums + 1)


nlis = [0] + list(map(int, input().split()))
ans = 0


def setbit(pos, diff):
    while pos <= nums+1:
        bit[pos] += diff
        pos += pos & -pos


def getbit(pos):
    new_ans = 0
    while pos:
        new_ans += bit[pos]
        pos -= pos & -pos
    return new_ans


for i in range(1, nums+1):
    setbit(i, nlis[i])


while True:
    try:
        char, number = input().split()
        number = int(number)

        match char:
            case "a":
                setbit(number, -nlis[number])
            case "?":
                ans = getbit(number-1)
                print(ans)

    except EOFError:
        break

